### PR TITLE
Rename tracing to telemetry package

### PR DESCRIPTION
**Pull request by pgomulka**
This commit renames the tracing to telemetry.tracing in both xpack/APM and elasticserach's org.elasticsearch.tracing.Tracer (the api)
the xpack/APM is renamed as follows:
`org.elasticsearch.telemetry.apm` - the only exported package
`org.elasticsearch.telemetry.apm.settings`  - `APMSettings`
`org.elasticsearch.telemetry.apm.tracing` - `APMTracer` 

`org.elasticsearch.tracing.Tracer` is moved to `org.elasticsearch.telemetry.tracing.Tracer` (responsible for majority of the changes in this PR)

### CODE REVIEW

**Review by juliaElastic — APPROVED** [review]
Fleet change LGTM

File: modules/apm/src/main/java/org/elasticsearch/telemetry/apm/settings/APMAgentSettings.java [hunk 1]
```diff
@@ -31,7 +32,7 @@
  * This class is responsible for APM settings, both for Elasticsearch and the APM Java agent.
  * The methods could all be static, however they are not in order to make unit testing easier.
  */
-class APMAgentSettings {
+public class APMAgentSettings {
```
Comment (stu-elastic) [inline modules/apm/src/main/java/org/elasticsearch/telemetry/apm/settings/APMAgentSettings.java:35]: Why the visibility changes for the class and methods?

File: modules/apm/src/main/java/org/elasticsearch/telemetry/apm/tracing/APMTracer.java [hunk 2]
```diff
@@ -98,7 +98,7 @@ public APMTracer(Settings settings) {
         this.enabled = APM_ENABLED_SETTING.get(settings);
     }
 
-    void setEnabled(boolean enabled) {
+    public void setEnabled(boolean enabled) {
```
Comment (stu-elastic) [inline modules/apm/src/main/java/org/elasticsearch/telemetry/apm/tracing/APMTracer.java:101]: Why change the visibility?

File: modules/apm/src/main/java/org/elasticsearch/telemetry/apm/settings/APMAgentSettings.java (hunk 1 above)
Comment (pgomulka) [reply modules/apm/src/main/java/org/elasticsearch/telemetry/apm/settings/APMAgentSettings.java:35]: it is the consequence of package structure.
org.elasticsearch.telemetry.apm - the only exported package - which requires that `APMTracer` goes to
org.elasticsearch.telemetry.apm.tracing. I don't want to have  `APMSettings` on the external package. So I moved it into its own package. 
As the `APMSettings` is no no longer in the same package as APMTracer I had to change the scope of methods.
It is safe that it is public, it is on a non exported package

File: modules/apm/src/main/java/org/elasticsearch/telemetry/apm/tracing/APMTracer.java (hunk 2 above)
Comment (pgomulka) [reply modules/apm/src/main/java/org/elasticsearch/telemetry/apm/tracing/APMTracer.java:101]: I moved APMTracer into its own package (I don't want it to be on an exported package). 
The setEnabled method is called from the APM class (on the exported package)

**Review by stu-elastic — APPROVED** [review]
lgtm

**Pull request merged**